### PR TITLE
 Add snapshot time

### PR DESCRIPTION
**Pull request by viko600**
Added snapshot creation time when the yb-admin list snapshot command is issuedalaigned

### CODE REVIEW

File: ent/src/yb/tools/yb-admin_client_ent.cc [hunk 1]
```diff
@@ -146,7 +147,8 @@ Status ClusterAdminClient::ListSnapshots(const ListSnapshotsFlags& flags) {
           HybridTimeToString(HybridTime::FromPB(entry.previous_snapshot_hybrid_time())),
           &json_snapshot, &document.GetAllocator());
     } else {
-      cout << SnapshotIdToString(snapshot.id()) << kColumnSep << snapshot.entry().state() << endl;
+      cout << SnapshotIdToString(snapshot.id()) << kColumnSep << snapshot.entry().state() << 
+        kColumnSep << HybridTimeToString(HybridTime::FromPB(snapshot.entry().snapshot_hybrid_time())) << endl;
```
Comment (sanketkedia) [inline ent/src/yb/tools/yb-admin_client_ent.cc:151]: For consistency, we generally align the `couts` like here - https://github.com/yugabyte/yugabyte-db/blob/273994915af69d5cc5086779d25a196ed6539f8d/src/yb/tools/yb-admin_client.cc#L1062

**Review by sanketkedia — CHANGES_REQUESTED** [review]
Hey Viktor, Great work on these changes. They look good. Left one minor comment. Also, it'd be great if you can attach a screenshot of how the `list_snapshots` command looks like both without and with your change since this is user-facing. 

Thank you!

File: ent/src/yb/tools/yb-admin_client_ent.cc [hunk 2]
```diff
@@ -123,7 +123,8 @@ Status ClusterAdminClient::ListSnapshots(const ListSnapshotsFlags& flags) {
   rapidjson::Value json_snapshots(rapidjson::kArrayType);
   if (!json) {
     if (resp.snapshots_size()) {
-      cout << RightPadToUuidWidth("Snapshot UUID") << kColumnSep << "State" << endl;
+      // Using 2 tabs so that the header can be aligned to the time
```
Comment (OlegLoginov) [inline ent/src/yb/tools/yb-admin_client_ent.cc:126]: Style:  '.' in the end of the comment.

File: ent/src/yb/tools/yb-admin_client_ent.cc [hunk 3]
```diff
@@ -123,7 +123,8 @@ Status ClusterAdminClient::ListSnapshots(const ListSnapshotsFlags& flags) {
...
+      cout << RightPadToUuidWidth("Snapshot UUID") << kColumnSep << "State" << kColumnSep << kColumnSep << "Creation Time" << endl;
```
Comment (OlegLoginov) [inline ent/src/yb/tools/yb-admin_client_ent.cc:127]: Style: please check that the line is not longer 100 characters.

File: ent/src/yb/tools/yb-admin_client_ent.cc (hunk 1 above)
Comment (OlegLoginov) [inline ent/src/yb/tools/yb-admin_client_ent.cc:151]: Style:  offset 4 spaces for a line continuation.

Comment (OlegLoginov) [inline ent/src/yb/tools/yb-admin_client_ent.cc:151]: Style: should be <= 100 characters.

**Review by OlegLoginov — CHANGES_REQUESTED** [review]

**Comment by viko600** [comment]
Hi @sanketkedia  and @OlegLoginov I have fixes the suggested errors and have made a screenshot with the 2 outputs of the `list_snapshots` command. For the first output I used my branch and for the second, I used the master branch.

changes vs current

File: ent/src/yb/tools/yb-admin_client_ent.cc [hunk 4]
```diff
@@ -123,7 +123,9 @@ Status ClusterAdminClient::ListSnapshots(const ListSnapshotsFlags& flags) {
   rapidjson::Value json_snapshots(rapidjson::kArrayType);
   if (!json) {
     if (resp.snapshots_size()) {
-      cout << RightPadToUuidWidth("Snapshot UUID") << kColumnSep << "State" << endl;
+      // Using 2 tabs so that the header can be aligned to the time.
+      cout << RightPadToUuidWidth("Snapshot UUID") << kColumnSep
```
Comment (sanketkedia) [inline ent/src/yb/tools/yb-admin_client_ent.cc:127]: Please remove trailing space

File: ent/src/yb/tools/yb-admin_client_ent.cc [hunk 5]
```diff
@@ -146,7 +148,9 @@ Status ClusterAdminClient::ListSnapshots(const ListSnapshotsFlags& flags) {
           HybridTimeToString(HybridTime::FromPB(entry.previous_snapshot_hybrid_time())),
           &json_snapshot, &document.GetAllocator());
     } else {
-      cout << SnapshotIdToString(snapshot.id()) << kColumnSep << snapshot.entry().state() << endl;
+      cout << SnapshotIdToString(snapshot.id()) << kColumnSep
```
Comment (sanketkedia) [inline ent/src/yb/tools/yb-admin_client_ent.cc:151]: Remove trailing space at the end of the line

File: ent/src/yb/tools/yb-admin_client_ent.cc [hunk 6]
```diff
@@ -146,7 +148,9 @@ Status ClusterAdminClient::ListSnapshots(const ListSnapshotsFlags& flags) {
           HybridTimeToString(HybridTime::FromPB(entry.previous_snapshot_hybrid_time())),
           &json_snapshot, &document.GetAllocator());
     } else {
-      cout << SnapshotIdToString(snapshot.id()) << kColumnSep << snapshot.entry().state() << endl;
+      cout << SnapshotIdToString(snapshot.id()) << kColumnSep 
+           << snapshot.entry().state() << kColumnSep
```
Comment (sanketkedia) [inline ent/src/yb/tools/yb-admin_client_ent.cc:152]: Remove trailing space

File: ent/src/yb/tools/yb-admin_client_ent.cc [hunk 7]
```diff
@@ -146,7 +148,9 @@ Status ClusterAdminClient::ListSnapshots(const ListSnapshotsFlags& flags) {
           HybridTimeToString(HybridTime::FromPB(entry.previous_snapshot_hybrid_time())),
           &json_snapshot, &document.GetAllocator());
     } else {
-      cout << SnapshotIdToString(snapshot.id()) << kColumnSep << snapshot.entry().state() << endl;
+      cout << SnapshotIdToString(snapshot.id()) << kColumnSep 
+           << snapshot.entry().state() << kColumnSep 
+           << HybridTimeToString(HybridTime::FromPB(snapshot.entry().snapshot_hybrid_time())) << endl;
```
Comment (sanketkedia) [inline ent/src/yb/tools/yb-admin_client_ent.cc:153]: The line is 102 characters long, please make it <= 100.

**Review by sanketkedia — CHANGES_REQUESTED** [review]
Hey @viko600, Some comments on code style, and then we are good to go.

File: ent/src/yb/tools/yb-admin_client_ent.cc (hunk 7 above)
Comment (viko600) [reply ent/src/yb/tools/yb-admin_client_ent.cc]: Sorry for that. Forgot to count the spaces in front of the line. Now should be fixed.

**Comment by viko600** [comment]
I want to take another issue after this one is resolved. Should I only look for issues with the `help wanted` label, or an I free to chose from all the issues?

**Review by OlegLoginov — APPROVED** [review]
Looks good to me.

**Comment by sanketkedia** [comment]
Backup tests are failing due to this change because they expect a strict format. I'll commit the change to fix them, run tests and then we'll probably be good to go.

**Comment by sanketkedia** [comment]
Screen Shot 2022-01-13 at 3 55 23 AM

Adding screenshot of the behavior when the snapshot is in "CREATING" or "DELETING" state.

**Review by sanketkedia — APPROVED** [review]
@viko600 Tests look good, so accepting this PR. Also, cases like the behavior when the snapshot is in "CREATING" or "DELETING" state also work. I will merge it.

**Comment by sanketkedia** [comment]
> I want to take another issue after this one is resolved. Should I only look for issues with the `help wanted` label, or an I free to chose from all the issues?

@viko600 We don't necessarily restrict from our end but the `help wanted` label is guidance on issues with less complicated fixes that are great for the OSS community. Also, do check with the issue creator/other people in the thread to see if somebody is already working on it. It could be that internally someone has already started working on it.

**Pull request merged**